resources/fetchersManager/data.go

```diff
@@ -50,68 +57,87 @@ func NewData(log *logp.Logger, interval time.Duration, timeout time.Duration, fe
 		timeout:    timeout,
 		interval:   interval,
 		fetchers:   fetchers,
-		stop:       make(chan struct{}),
-		stopNotice: make(chan time.Duration),
+		stopNotice: make(chan time.Duration, 1),
 	}, nil
 }
 
+var ErrGracePeriodTooLong = errors.New("graceful shutdown too long")
+var ErrGracePeriodEnded = errors.New("graceful shutdown ended")
```

Comment:
depending on why we exit, we cancel with a different `cause`. this is needed for tests to assert on why we exited. otherwise we wouldn't bother with it.